src/libstd/sys/unix/fs.rs

```diff
@@ -108,6 +108,19 @@ impl FilePermissions {
     pub fn mode(&self) -> raw::mode_t { self.mode }
 }
 
+/// OS-specific extension methods for `fs::FileType`
+#[unstable(feature = "file_type_ext", reason = "recently added API")]
+pub trait FileTypeExt {
+    /// Gain a reference to the underlying `mode` data which contains the
+    /// raw information returned by the OS.
+    fn raw_mode(&self) -> mode_t;
+}
+
+#[unstable(feature = "file_type_ext", reason = "recently added API")]
+impl FileTypeExt for ::fs::FileType {
+    fn raw_mode(&self) -> mode_t { self.as_inner().mode }
+}
```

Comment:
You can probably actually avoid this trait impl entirely and instead just making the `is` method below public and updating the impl in `ext/fs.rs` to do something like:

```rust
pub fn is_foo(&self) -> bool { self.as_inner().is(libc::S_IFFOO) }
```